internal/campaigns/repo_fetcher.go

```diff
@@ -80,16 +113,19 @@ func (rf *repoFetcher) Fetch(ctx context.Context, repo *graphql.Repository) (Rep
 		}
 	}
 
-	return &repoZip{
-		path:    path,
-		fetcher: rf,
-	}, nil
+	zip.references += 1
+	return zip, nil
 }
 
 func (rz *repoZip) Close() error {
-	if rz.fetcher.deleteZips {
+	rz.mu.Lock()
+	defer rz.mu.Unlock()
+
+	rz.references -= 1
+	if rz.references == 0 && rz.fetcher.deleteZips {
```

Comment:
Wouldn't this mean when using 1 worker thread, there would be no caching at all? and can there be the case where the workers are utilized like the following

```
[repo A:/path1]
[repo B]
[repo C]
[repo D]

=>

[repo A:/path2]
[repo B]
[repo C]
[repo D]
```

where the worker would close the zip for repo A at path1, and then need to refetch for path2?

Reply:
Nice catch. You're right. I'll address in follow-up PR.